pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -0,0 +1,199 @@
+	"k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+	nodeutil "sigs.k8s.io/descheduler/pkg/descheduler/node"
+	podutil "sigs.k8s.io/descheduler/pkg/descheduler/pod"
+	"sigs.k8s.io/descheduler/pkg/framework"
+	"sigs.k8s.io/descheduler/pkg/utils"
+)
+
+const (
+	PluginName            = "DefaultEvictor"
+	evictPodAnnotationKey = "descheduler.alpha.kubernetes.io/evict"
+)
+
+var _ framework.EvictorPlugin = &DefaultEvictor{}
+
+type constraint func(pod *v1.Pod) error
+
+// DefaultEvictor is the first EvictorPlugin, which defines the default extension points of the
+// pre-baked evictor that is shipped.
+// Even though we name this plugin DefaultEvictor, it does not actually evict anything,
+// This plugin is only meant to customize other actions (extension points) of the evictor,
+// like filtering, sorting, and other ones that might be relevant in the future
+type DefaultEvictor struct {
+	args        *DefaultEvictorArgs
+	constraints []constraint
+}
+
+// IsPodEvictableBasedOnPriority checks if the given pod is evictable based on priority resolved from pod Spec.
+func IsPodEvictableBasedOnPriority(pod *v1.Pod, priority int32) bool {
+	return pod.Spec.Priority == nil || *pod.Spec.Priority < priority
+}
+
+// HaveEvictAnnotation checks if the pod have evict annotation
+func HaveEvictAnnotation(pod *v1.Pod) bool {
+	_, found := pod.ObjectMeta.Annotations[evictPodAnnotationKey]
+	return found
+}
+
+type DefaultEvictorOptions struct {
+	Priority      *int32
+	NodeFit       bool
+	LabelSelector labels.Selector
+}
+
+// New builds plugin from its arguments while passing a handle
+func New(args runtime.Object, handle framework.Handle) (framework.Plugin, error) {
+	defaultEvictorArgs, ok := args.(*DefaultEvictorArgs)
+	if !ok {
+		return nil, fmt.Errorf("want args to be of type defaultEvictorFilterArgs, got %T", args)
+	}
+	ev := &DefaultEvictor{
+		args: defaultEvictorArgs,
+	}
+
+	if ev.args.EvictFailedBarePods {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			ownerRefList := podutil.OwnerRef(pod)
+			// Enable evictFailedBarePods to evict bare pods in failed phase
+			if len(ownerRefList) == 0 && pod.Status.Phase != v1.PodFailed {
+				return fmt.Errorf("pod does not have any ownerRefs and is not in failed phase")
+			}
+			return nil
+		})
+	} else {
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			ownerRefList := podutil.OwnerRef(pod)
+			if len(ownerRefList) == 0 {
+				return fmt.Errorf("pod does not have any ownerRefs")
+			}
+			return nil
+		})
+	}
+	if !ev.args.EvictSystemCriticalPods {
+		klog.V(1).InfoS("Warning: EvictSystemCriticalPods is set to True. This could cause eviction of Kubernetes system pods.")
+		ev.constraints = append(ev.constraints, func(pod *v1.Pod) error {
+			if utils.IsCriticalPriorityPod(pod) {
+				return fmt.Errorf("pod has system critical priority")
+			}
+			return nil
+		})
+
+		if ev.args.PriorityThreshold != nil && ev.args.PriorityThreshold.Value != nil {
```

Comment:
You can drop `&& ev.args.PriorityThreshold.Value != nil` as `GetPriorityValueFromPriorityThreshold` falls back to `.Name` if `.Value` is nil. Resp. you can `&& (ev.args.PriorityThreshold.Value != nil || len(ev.args.PriorityThreshold.Name) > 0)`

Reply:
You mean 

instead of

```
if ev.args.PriorityThreshold != nil && ev.args.PriorityThreshold.Value != nil
```
we check for
```
if defaultEvictorArgs.PriorityThreshold != nil && (defaultEvictorArgs.PriorityThreshold.Value != nil || len(defaultEvictorArgs.PriorityThreshold.Name) > 0)
```

?

That's what I did now :heavy_check_mark: